pkg/reconciler/tenancy/workspacemounts/workspacemounts_reconcile_updater.go

```diff
@@ -85,52 +61,50 @@ func (r *workspaceStatusUpdater) reconcile(ctx context.Context, workspace *tenan
 			)
 			return reconcileStatusContinue, nil
 		}
+		return reconcileStatusStopAndRequeue, err
+	}
 
-		// we are working on status field. As this is "loose coupling, we parse it out"
-		// Mount point implementors must expose status.{URL,phase} as a fields.
-		// We are not interested in the rest of the status.
-		statusPhase, ok, err := unstructured.NestedString(obj.Object, "status", "phase")
-		if err != nil || !ok {
-			conditions.MarkFalse(
-				workspace,
-				tenancyv1alpha1.MountConditionReady,
-				tenancyv1alpha1.MountObjectNotReadyReason,
-				conditionsv1alpha1.ConditionSeverityError,
-				"Mount is not reporting ready. See %s %q status for more details",
-				obj.GroupVersionKind().Kind, obj.GetName(),
-			)
-
-			return reconcileStatusContinue, nil //nolint:nilerr // we ignore the error intentionally. Not helpful.
-		}
-
-		// url might not be there if the mount is not ready
-		statusURL, _, _ := unstructured.NestedString(obj.Object, "status", "URL")
+	// we are working on status field. As this is "loose coupling, we parse it out"
+	// Mount point implementors must expose status.{URL,phase} as a fields.
+	// We are not interested in the rest of the status.
+	statusPhase, ok, err := unstructured.NestedString(obj.Object, "status", "phase")
+	if err != nil || !ok {
+		conditions.MarkFalse(
+			workspace,
+			tenancyv1alpha1.MountConditionReady,
+			tenancyv1alpha1.MountObjectNotReadyReason,
+			conditionsv1alpha1.ConditionSeverityError,
+			"Mount is not reporting ready. See %s %q status for more details",
+			obj.GroupVersionKind().Kind, obj.GetName(),
+		)
 
-		// Only Spec or Status can be updated, not both.
-		if mount.MountStatus.Phase != tenancyv1alpha1.MountPhaseType(statusPhase) || mount.MountStatus.URL != statusURL {
-			mount.MountStatus.Phase = tenancyv1alpha1.MountPhaseType(statusPhase)
-			mount.MountStatus.URL = statusURL
+		return reconcileStatusContinue, nil //nolint:nilerr // we ignore the error intentionally. Not helpful.
+	}
 
-			workspace.Annotations[tenancyv1alpha1.ExperimentalWorkspaceMountAnnotationKey] = mount.String()
+	// url might not be there if the mount is not ready
+	statusURL, _, _ := unstructured.NestedString(obj.Object, "status", "URL")
```

Comment:
can we pull the workspaceType and Cluster too? otherwise adopting this against code that already exists would be pretty hard. the other option is to relax admission with the ability to set those when mount is specified

Reply:
nvm, looks like validation is relaxed, still would really be nice to pull the type and cluster back. otherwise all existing integrations would have to be reworked. that could slow/prevent adoption of mounts